log-replication/src/main/java/org/corfudb/logreplication/infrastructure/CorfuInterClusterReplicationServer.java

```diff
@@ -46,8 +46,8 @@
                     + "\n"
                     + "Usage:\n"
                     + "\tlog_replication_server [-a <address>|-q <interface-name>] "
-                    + "[-c <ratio>] [-d <level>] [-p <seconds>] "
-                    + "[--base-server-threads=<base_server_threads>] "
+                    + "[-c <ratio>] [-d <level>] [-p <seconds>] [--site-manager-config <config>]"
```

Comment:
My understanding is that you are adding the option of passing a file that has a fixed topology configuration, right? ... To avoid adding more flags that are merely for testing, probably what is cleaner is the following:

- We have the flag --plugin, which intention is to specify the location of a file that contains  the class path to the external "adapters", which includes site manager adapter and other plugins (transport and configuration)

So the idea is that there you specify your test adapter. Let's say you create a class that you call UnitTestClusterManagerAdapter. This class will always represent your adapter for testing, and now that class reads from a fixed location the configuration (instead of really querying some external entity), And that such file with the config is this file you are trying to pass. 

If you want to change config in between tests, your adapter is the same, you only need to modify the content that the config is retrieving. That way we avoid bringing in running parameters that are test specific.